Reduce onboarding steps

jcapona opened the pull request:
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/SWE-363) |


#### Main changes
- Remove Terms and Keyboard pages.
- Include terms in Registration Page, displayed in a dialog when clicking the text.
- Don't upgrade the system during onboarding, only the web-portal.

#### Screenshots (feature, test output, profiling, dev tools etc)

New registration page:
![image](https://github.com/user-attachments/assets/fbf97523-9a24-4ac4-bb63-0f6a42b55512)

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

jcapona committed:
Remove Terms and Keyboard pages

jcapona committed:
Include terms in Registration Page

jcapona committed:
Dont upgrade system on onboarding

angusjfw commented:
Can we change the  update page title - people get confused that it says 'Ok, I **need** to be updated' and then sometimes there aren't any updates. Instead say "Ok, checking for updates..."

angusjfw commented:
the update page no longer needs the 'update' button since it's always disabled - this page should now just run automatically and then move on

jcapona committed:
Safely check for wlan_ap0 ip to determine if should switch networks

jcapona committed:
Don't display 'next/update' button during upgrade

jcapona committed:
Add cache control headers to index.html and backend response

- Added meta tags for cache control in index.html to prevent caching.
- Implemented a no-cache response header in the Flask app to ensure fresh content delivery.

jcapona committed:
Update copy when preparing web-portal

jcapona committed:
Centralize axios instance and enforce no-cache policy

jcapona committed:
Add thread safety to build-info route

- Added a threading lock to ensure safe access to the '/build-info' route, preventing concurrent requests from causing locks when trying to read from the hub.

jcapona committed:
Update web-portal and its dependencies
